File: java-suite/src/main/java/de/fraunhofer/iem/functions/simpleExample/classMemberFunctions/TwoInts.java
package de.fraunhofer.iem.functions.simpleExample.classMemberFunctions;

/**
 * Demonstrates the non-static member functions in Java
 *
 * @author Ranjith Krishnamurthy
 */
public class TwoInts {
    /**
     * Demonstrates the public non-static member function in Java
     */
    public int add(int num1, int num2) {
        return num1 + num2;
    }

    /**
     * Demonstrate the private non-static member function in Java
     */
    private float add10(int num1, int num2) {
        return ((float) num1) + num2 + 10;
    }

    public float publicAdd10(int num1, int num2) {
        return add10(num1, num2);
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/functions/simpleExample/classMemberFunctions/TwoInts.kt
package de.fraunhofer.iem.functions.simpleExample.classMemberFunctions

class TwoInts() {
    /**
     * Demonstrate the public class member function in Kotlin
     *
     * @author Ranjith Krishnamurthy
     */
    fun add(num1: Int, num2: Int): Int = num1 + num2

    fun publicAdd10(num1: Int, num2: Int) = add10(num1, num2)
    
    /**
     * Demonstrate the private class member function in Kotlin
     *
     * @author Ranjith Krishnamurthy
     */
    private fun add10(num1: Int, num2: Int): Float {
        return num1.toFloat() + num2 + 10
    }

    /**
     * Demonstrate internal class member single-expression function in Kotlin
     *
     * @author Ranjith Krishnamurthy
     */
    internal fun sub(num1: Int, num2: Int) = num1 - num2
}